include/boost/decimal/detail/cmath/riemann_zeta.hpp

```diff
@@ -219,7 +219,7 @@ constexpr auto riemann_zeta(T x) noexcept
 
 BOOST_DECIMAL_EXPORT template <typename T, typename IntegralType>
 constexpr auto riemann_zeta(IntegralType n) noexcept
-    BOOST_DECIMAL_REQUIRES_TWO(detail::is_decimal_floating_point_v, T, std::is_integral_v, IntegralType)
+    -> typename std::enable_if<detail::is_decimal_floating_point_v<T> && std::is_integral<IntegralType>::value, T>::type
```

Comment:
`BOOST_DECIMAL_REQUIRES_TWO` returns `promote_args_t<T1, T2>`. Here you would want `BOOST_DECIMAL_REQUIRES_TWO_RETURN` which has args constraint1, type1, constraint2, type2, ReturnType.